deduce.go

```diff
@@ -690,6 +695,253 @@ func (sm *SourceMgr) deduceFromPath(path string) (deductionFuture, error) {
 	}, nil
 }
 
+type deductionCoordinator struct {
+	ctx      context.Context
```

Comment:
I don't really follow most of that discussion, but at the very least, this is unusual, which in the Go community is a strong knock against it :)

It appears that the stored context is only really used in `deduceRootPath` to check whether the coordinator has already been terminated. The fact that it's done with a stored context (but other times a context is passed in to the function call - why? what's the difference?) will make it harder to see _how_ you go about canceling calls, I think. I could be wrong! There might be more inherent complexity here than I see, so maybe that difficulty is going to be here no matter what.

But I agree with Peter that I've regretted doing things this way every time. Letting the context flow down through method invocations has never confused me in the same way.